tests/pos/i19266.scala

```diff
@@ -0,0 +1,9 @@
+object i19266:
```

Comment:
note:

```
  def fn1(x: Int, y: String = "y")(z: Double) = Some(s"$x$y$z")
  def fn2(p: Int)(q: String) = Some(s"$p$q")

  def checkCompile =
    // This is NOT regarded as `isPureExpr`, but I think should be to warn `nonunit-statement`.
    // See https://github.com/lampepfl/dotty/issues/19266#issuecomment-1856248380 
    fn1(1)
    // This is regarded as `isPureExpr` and therefore it results in error.
    fn2(2)
```

Reply:
```
Block(
    List(
        ValDef(
            y$1,
            TypeTree[AnnotatedType(TypeRef(TermRef(ThisType(TypeRef(NoPrefix,module class scala)),object Predef),type String),ConcreteAnnotation(Apply(Select(New(TypeTree[TypeRef(ThisType(TypeRef(NoPrefix,module class unchecked)),class uncheckedVariance)]),<init>),List())))],
            Select(This(Ident(i19266$)),fn1$default$2)
        )
    ),
    Block(
        List(
            DefDef(
                $anonfun,
                List(List(ValDef(z,TypeTree[TypeRef(TermRef(ThisType(TypeRef(NoPrefix,module class <root>)),object scala),class Double)],EmptyTree))),
                TypeTree[AppliedType(TypeRef(ThisType(TypeRef(NoPrefix,module class scala)),class Some),List(TypeRef(ThisType(TypeRef(NoPrefix,module class lang)),class String)))],
                Apply(Apply(Ident(fn1),List(Literal(Constant(1)), Ident(y$1))),List(Ident(z)))
            )
        ),
        Closure(List(),Ident($anonfun),EmptyTree)
    )
)
Block(
    List(
        DefDef(
            $anonfun,
            List(List(ValDef(q,TypeTree[TypeRef(TermRef(ThisType(TypeRef(NoPrefix,module class scala)),object Predef),type String)],EmptyTree))),
            TypeTree[AppliedType(TypeRef(ThisType(TypeRef(NoPrefix,module class scala)),class Some),List(TypeRef(ThisType(TypeRef(NoPrefix,module class lang)),class String)))],
            Apply(Apply(Ident(fn2),List(Literal(Constant(2)))),List(Ident(q)))
        )
    ),
    Closure(List(),Ident($anonfun),EmptyTree)
)
```

Reply:
`f1(1)` is regarded as Block with `ValDef` statement and `Block` expr whereas `f2(2)` is regarded as `Block` with `DefDef` statement and `Closure`.